modules/flowable-ui-idm/flowable-ui-idm-rest/src/main/java/org/flowable/ui/idm/rest/app/IdmSSOResource.java

```diff
@@ -0,0 +1,181 @@
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.google.common.net.HttpHeaders;
+
+/**
+ * @author Matthias De Bie
+ * <p>
+ * TODO: update user before logging in
+ * TODO: make a config option to automatically redirect to sso in stead of idm
+ */
+@RestController
+@RequestMapping("/app")
+public class IdmSSOResource {
+
+    @Autowired
+    protected UserService userService;
+
+    @Autowired
+    protected PrivilegeService privilegeService;
+
+    @Autowired
+    protected SSOHandler ssoHandler;
+
+    @Autowired
+    protected RememberMeServices rememberMeServices;
+
+    @RequestMapping(value = "/sso/external", method = RequestMethod.GET)
+    public void getSsoExternalUrl(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        String urlBase = request.getRequestURL().toString().replace("/sso/external", "/sso/return");
+
+        if (urlBase.startsWith("http://") && "https".equals(request.getHeader(HttpHeaders.X_FORWARDED_PROTO))) {
+            urlBase = urlBase.replace("http://", "https://");
+        }
+
+        if (ssoHandler.isActive()) {
+            response.getWriter().write(
+                ssoHandler.getExternalUrl(urlBase)
+            );
+            response.flushBuffer();
+        }
+    }
+
+    @RequestMapping(value = "/sso/return", method = RequestMethod.GET)
+    public void onSsoReturnGet(HttpServletRequest request, HttpServletResponse response) {
+        handleSSOReturn(request, response, null);
+    }
+
+    @RequestMapping(value = "/sso/return", method = RequestMethod.POST)
+    public void onSsoReturn(@RequestBody MultiValueMap<String, String> body, HttpServletRequest request, HttpServletResponse response) {
+        handleSSOReturn(request, response, body);
+    }
+
+    private void handleSSOReturn(HttpServletRequest request, HttpServletResponse response, MultiValueMap<String, String> body) {
+        SSOUserInfo ssoUserInfo = ssoHandler.handleSsoReturn(request, response, body);
+
+        if (ssoUserInfo != null) {
+            try {
+                UserInformation userInfo = userService.getUserInformation(ssoUserInfo.getId());
+                loginWithSso(request, response, userInfo.getUser());
+
+            } catch (NotFoundException e) {
+                // userInfo not found: create new account
+                createSsoAccount(request, response, ssoUserInfo);
+
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            response.setStatus(HttpStatus.UNAUTHORIZED.value());
+        }
+    }
+
+    private static Logger LOGGER = LoggerFactory.getLogger(IdmSSOResource.class);
+
+    private void loginWithSso(HttpServletRequest request, HttpServletResponse response, User user) throws IOException {
+        String redirectTo = request.getRequestURL().toString().replace("/app/sso/return", "");
```

Comment:
When you get redirected to the login page, the page saves the query parameter `redirectUri` (the one that tells the idm where to go after successful login). Because you would lose this parameter when going to the SSO, I saved it in the cookies. This was again the only way I found to work to save the redirectPath.

The replace was done to get the baseURL. `<fullurl>/app/sso/return`*.replace("/app/sso/return", "")* = the full url.
This is indeed a dirty hack, if you know how to get it properly, please let me know.